lte/gateway/python/scripts/user_trace_cli.py

```diff
@@ -0,0 +1,302 @@
+            help="IMSI of the subscriber",
+        )
+        session_trace.set_defaults(func=self.get_session_trace)
+
+        return parser
+
+    def get_list_imsi(self, args):
+        """List imsi and number of ocurrences in logs"""
+
+        imsi_pattern = r"\b\d{15}\b"
+
+        imsi_list = defaultdict(int)
+        lines = self.lines
+        for line in lines:
+            match = re.findall(imsi_pattern, line)
+            if not match:
+                continue
+            imsi = match[0]
+            imsi_list[imsi] += 1
+
+        sorted_imsi_list = sorted(imsi_list.items(), key=lambda x: x[1], reverse=True)
+
+        print("\n{0:<15} {1:<15} ".format('IMSI', 'Occurrences'))
+        print('-' * 30)
+
+        for v in sorted_imsi_list:
+            imsi, number = v
+            print("{0:<15} {1:<15}".format(imsi, number))
+
+        print("\nNumber of unique IMSI found in logs: " + str(len(imsi_list)))
+
+    def get_list_ue_id(self, args):
+        """
+        Provide list of mme id and enodeb id pairs from given an IMSI.
+        Some sessions may not have the pair(missing enodeb id). Getting mme_id from Attach Req
+
+        Args:
+            args: Filtered logs for a given imsi and mme id
+
+        Returns:
+            ue_id_list: list of mme id and enodeb id pairs from given an IMSI
+        """
+
+        hex_pattern = "0X.*"
+        pattern_mme_enb_ue_id = r".*MME_UE_S1AP_ID = (\w+)\b eNB_UE_S1AP_ID = (\w+)\b"
+        ue_id_list = []
+        mme_id_list = []
+        lines = self.lines
+        imsi = self.args.imsi
+        for line in lines:
+            ue_id = {}
+            match = re.findall(imsi + pattern_mme_enb_ue_id, line, re.IGNORECASE)
+
+            if not match:
+                continue
+
+            ue_id["mme_ue_id"] = match[0][0]
+            ue_id["enb_ue_id"] = match[0][1]
+
+            if re.search(hex_pattern, match[0][0], re.IGNORECASE) and re.search(hex_pattern, match[0][1], re.IGNORECASE):
+                ue_id["mme_ue_id"] = int(ue_id["mme_ue_id"], 16)
+                ue_id["enb_ue_id"] = int(ue_id["enb_ue_id"], 16)
+
+            if ue_id and ue_id not in ue_id_list:
+                ue_id_list.append(ue_id)
+                mme_id_list.append(ue_id["mme_ue_id"])
+
+        # Some sessions may not have enodeb id, getting the mme_id from Attach request
+
+        attach_pattern = r"ATTACH REQ \(ue_id = (\w+)\) \(IMSI = " + imsi + r"\)"
+
+        for line in lines:
+            match = re.findall(attach_pattern, line)
+            if not match:
+                continue
+            mme_id = int(match[0], 16)
+            if mme_id not in mme_id_list:
+                mme_id_list.append(mme_id)
+                ue_id_list.append({"mme_ue_id": mme_id, "enb_ue_id": "None"})
+
+        print("\nIMSI: " + imsi + "\n")
+        print("{0:<15} {1:<15} ".format('mme_id', 'enodeb_id'))
+        print('-' * 25)
+
+        for v in ue_id_list:
+            mme, enodeb = v.items()
```

Comment:
if we don't append `enb_ue_id` on line 161, we can update the check to:
```suggestion
            mme_ue_id = v.get('mme_id')
            enodeb_ue_id = v.get('enb_ue_id')
            if not enodeb_ue_id:
                print...
            else:
               print...
            
```

Reply:
great suggestion! thank you, I have updated with the comments